docs/_source/practical_guides/annotation_workflows/semantic_search.md

```diff
@@ -0,0 +1,361 @@
+# 🔦 Semantic Search
+
+This guide gives an overview of the semantic search features. Since `1.19.0` Argilla supports adding vectors to Feedback Datasets (other datasets include this feature since `1.2.0`) which can then be used for finding the most similar records to a given one. This feature uses vector or semantic search combined with more traditional search (keyword and filter-based).
+
+Vector search leverages machine learning to capture rich semantic features by embedding items (text, video, images, etc.) into a vector space, which can be then used to find "semantically" similar items.
+
+In this guide, you'll find how to:
+
+- Set up your Elasticsearch or Opensearch endpoint with vector search support.
+- Encode text into vectors for Argilla records.
+- Use semantic search.
+
+The next section gives a general overview of how semantic search works in Argilla.
+
+## How it works
+
+Semantic search in Argilla works as follows:
+
+1. One or several vectors can be included in the `vectors` field of Argilla Records. The `vectors` field accepts a dictionary where `keys` represent the names and `values` contain the actual vectors. This is the case because certain use cases might require using several vectors. Note that for a `FeedbackDataset` you will also need to [configure `VectorSettings`](/practical_guides/create_dataset.md#define-vectors) in your dataset.
+2. The vectors are stored at indexing time, once the records are logged with `add_records` or `update_records` in a `FeedbackDataset`, or with `rg.log` in older datasets.
+3. If you have stored vectors in your dataset, you can use the semantic search feature in Argilla UI and the Python SDK.
+
+In future versions, embedding services might be developed to facilitate steps 1 and 2 and associate vectors to records automatically.
+
+```{note}
+
+It's completely up to the user which encoding or embedding mechanism to use for producing these vectors. In the "Encode text fields" section of this document you will find several examples and details about this process, using open-source libraries (e.g., Hugging Face) as well as paid services (e.g., Cohere or OpenAI).
+
+Currently, Argilla uses vector search only for searching similar records (nearest neighbors) of a given vector. This can be leveraged from Argilla UI as well as the Python Client. In the future, vector search could be leveraged as well for free text queries using Argilla UI.
+
+```
+
+## Setup vector search support
```

Comment:
I think this might be a bit overkill. We already have ES and OS enabled by default before we can actually annotate, so just mentioning the migration to a potential newer version would be fine IMO.